### ISSUE TITLE
update_cdf semantics

### ISSUE THREAD

**Issue by cconcolato** [AOMediaCodec/av1-spec#2]
The spec says:
"update_cdf is a function call that indicates that the CDF arrays are set equal to the final CDFs at the end of the largest tile (measured in bytes). This process is described in section 7.4."

However, the update_cdf() function is called only once in the last tile not the largest:
```
if (tg_end == NumTiles - 1) {
        if ( !error_resilient_mode && !frame_parallel_decoding_mode ) {
            update_cdf( )
        }
...
}
```

Should the semantics or the syntax be fixed?

**Comment by peterderivaz** [comment]
As each tile is decoded it calls init_bool and exit_bool.

init_bool sets the variable UpdateCdfs to indicate if this is the biggest tile seen so far.

exit_bool copies the tile CDFs into SavedCDFs if UpdateCdfs is set.  (This may happen multiple times in a frame if tile sizes increase.)

When the whole frame is decoded (i.e. when we have just decoded the last tile), we call update_cdf.

This process will copy the SavedCDFs into the frame CDF arrays.

In summary, we want to call exit_bool once per tile, but update_cdf once per frame.

**Comment by cconcolato** [comment]
Thanks. It's clearer. I would recommend rephrasing a bit `update_cdf` semantics then to something like:
> update_cdf is a function call that indicates that the CDF arrays are set equal to the final CDFs at the end of a frame, based on CDF arrays updated at the end of the largest tile (measured in bytes). This process is described in section 7.4.

**Comment by peterderivaz** [comment]
I've updated in line with your suggestion here:

https://github.com/AOMediaCodec/av1-spec/commit/f93dcb58fe9585d4d6a2cab5260afea5070b45f6
